index.bs

```diff
@@ -253,18 +253,251 @@ Control Protocols {#control}
 Issue(77): Describe CBOR based mechanism for
 exchanging control messages, results, and events.
 
-Presentation API Protocol {#presentation-api}
+
+Presentation Protocol {#presentation-protocol}
 ---------------------------------------------
 
-Issue(55): Incorporate material from the
-[Protocol](protocol.md) document.
+This section defines the use of the Open Screen Protocol for starting,
+stopping, and controlling presenetations as defined by
+[[PRESENTATION-API|Presentation API]].  A subsequent section will
+define how APIs in [[PRESENTATION-API|Presentation API]] map to the
+protocol messages defined in this section.
+
+For all messages defined in this section, see Appendix A for the full
+CDDL definitions.
+
+<!-- TODO: Add a capability that indicates support for the
+presentation protocol -->
+
+To learn which receivers are [=available presentation display=]s for a
+particular URL or set of URLs, the controller may send a
+presentation-url-availability-request message with the following values:
+
+- urls: A list of presentation URLs
+
+- watch-duration: The period of time that the controller is interested in
+receiving updates about the URLs, should the availability change.
+
+- watch-id: An identifier the receiver may use when sending updates about URL
+availability so that controller knows which URLs the receiver is referring to.
+
+In response, the receiver should send one presentation-url-availability-response
+message with the following values:
+
+- url-availabilities: A list of URL availability states (available, unavailable,
+or invalid).  Each state must correspond to the matching URL from the request by
+list index.
+
+
+The receivers should later (up to the current time plus request
+watch-duration) send presentation-url-availability-event messages if
+URL availabilities change.  Such events contain the following values:
+
+- watch-id: The watch-id given in the presentation-url-availability-response,
+used to refer to the presentation URLs whose availability has changed.
+
+- url-availabilities: A list of URL availability states (available, unavailable,
+or invalid).  Each state must correspond to the URLs from the request referred
+to by the watch-id.
+
+Note that these messages are not broadcasted to all controllers. They are sent
+individually to controllers that have requested availability for the URLs that
+have changed in availability state within the watch duration of the original
+availability request.
```

Comment:
On the other hand, it may be useful for a controller to be able to update a previous watch (URL list). Otherwise, each time a new URL availability is required, it will have to register a new watch  including all previous URLs, and get duplicate updates for them.

Reply:
It can just not renew watches and keep relatively short durations for them.